### ISSUE TITLE
Model containing Vec3 (from elm-explorations/linear-algebra) causes parsing to fail

### ISSUE THREAD

**Issue by MartinSStewart** [layflags/elm-monitor#11]
The following code will generate an error in the console when clicking the button. Note that the error is not generated if Vec3 is removed from the model.
```
module Main exposing (main)

import Math.Vector3 as Vec3
import Monitor
import Html exposing (Html)
import Html.Events

type Msg_
    = Msg_ 


type alias Model_ =
    { v : Vec3.Vec3
    }


main : Program () Model_ Msg_
main =
    Monitor.sandbox
        { init =
            { v = Vec3.vec3 0 0 0
            }
        , view = \_ -> Html.button [ Html.Events.onClick Msg_ ] [ Html.text "produces an error in console" ]
        , update = \_ model -> model
        }
```

**Comment by ChristophP** [comment]
I just looked and saw that the string representation of `Vec3` is the following.
```elm
> Debug.toString <| Math.Vector3.vec3 0 1 2
"{ 0 = 0, 1 = 1, 2 = 2 }" : String
```
So vectors seem to get special treatment in that regard. That output cannot be parsed with the record parser (because their fields aren't allowed to start with a digit) and they also don't look like a regular union type.

First, this made me realize that all the types in `linear-algebra` need to be special cases:
```elm
Vec2, Vec3, Vec4, Matrix4
```
Secondly that made me realize that there are potentially more cases we aren't covering right now. All of these would need special handling during parsing. Some things like `Regex` or so (unlikely to be in a model but still). Regex weirdly looks like "{}" when run through `Debug.toString`. Not sure what else we are missing.

I guess I can implement parsing for the types in `linear-algrebra`. That should cover those cases at least.

**Comment by layflags** [comment]
Thanks for reporting @MartinSStewart  ... I published the bugfix with v0.0.11

**Comment by layflags** [comment]
@ChristophP I just updated the parser to allow digits for object keys
